doltpy/core/write/write.py

```diff
@@ -85,3 +86,165 @@ def _import_helper(repo: Dolt,
     write_import_file(fp.name)
     args = ['table', 'import', table_name, '--pk={}'.format(','.join(primary_keys))] + import_flags
     repo.execute(args + [fp.name])
+
+
+def import_dict(repo: Dolt,
+                table_name: str,
+                data: Mapping[str, List[Any]],
+                primary_keys: List[str] = None,
+                import_mode: str = None):
+    """
+    COLUMN MAJOR (note this sort of takes care of list of list column major since user would have to provide such a function with column information anyway)
+
+    Import a dict mapping keys to lists of values for the rows. All lists must be of identical length. Suppose we have a table as follows:
+        CREATE TABLE players (id INT, name VARCHAR(16), PRIMARY KEY (id))
+
+    >>> dict_of_lists = {'id': [1, 2], 'name': ['Roger', 'Rafael']}
+    >>> import_dict(repo, 'players', dict_of_lists, import_mode='update')
+
+    Note that we can run this in create mode using the
+
+    Assertions:
+        - all list values are of equal length
+        - each list value has elements that can be cast to a common supertype that can be cast to the type of the column whose name the key to the list value
+
+    """
+    assert import_mode in [UPDATE, CREATE]
+
+    server_was_running = False
+    if not repo.server:
+        repo.sql_server()
+    else:
+        server_was_running = True
+
+    # Grab some basic information about the data
+    assert data, 'Cannot provide an empty dictionary'
+    row_count = len(list(data.values())[0])
+    assert row_count > 0, 'Must provide at least a single row'
+    assert all(len(val_list) == row_count for val_list in data.values()), 'Must provide value lists of uniform length'
+
+    # If the table does not exist, create it using type inference to build a create statement
+    if import_mode == CREATE:
+        _create_table_inferred(repo, table_name, data, primary_keys)
+
+    # Now transform the data to lists of tuples, where the elements are in the same order as the
+    # columns when sorted lexicographically
+    sorted_cols = sorted(data.keys())
+    tuple_list = []
+    for i in range(row_count):
+        tuple_list.append(tuple([data[col][i] for col in sorted_cols]))
+
+    insert_statement = _get_insert_statement(table_name, sorted_cols)
+    logger.info('Inserting {row_count} rows into table {table_name}'.format(row_count=row_count,
+                                                                            table_name=table_name))
+    conn = repo.get_connection()
+    cursor = conn.cursor()
+    cursor.executemany(insert_statement, tuple_list)
+    conn.commit()
+
+    if not server_was_running:
+        repo.sql_server_stop()
+
+
+def _create_table_inferred(repo: Dolt,
```

Comment:
I see the need for something that can infer types and import data from python data structures, but I'm not sure this is the right way to go about it. A more tested route would be to create the a csv file from the data and use `dolt table import -c table_name tmp.csv`. (Table import -c uses type inference).

If you want to keep these SQL statements for now, you should escape the table names with backticks to prevent errors. Beware, there are other gotchas that we've had to handle when generating SQL in the Dolt repo.